File: one_sample.c
/* Nathan Englehart (Summer, 2022) */

#include <stdio.h>
#include <math.h>

double one_sample(double x_bar, double mu, double s, int n) {

	/* Returns t value for one tailed t-test using given variables. */

	return (x_bar - mu) / (s / sqrt(n));

}

int main() {

	double x_bar = 74.0;
	double mu = 78.0;
	double s = 3.5;
	int n = 10;

	double t = fabs(one_sample(x_bar, mu, s, n));

	printf("absolute t = %f.\n",t);
	printf("df = %d.\n",n-1);

	return 0;

}

File: one_sample.r
#!/usr/bin/env Rscript

### Nathan Englehart (Summer 2022)

library(haven) # to read sav

gss_data <- read_sav("data.sav")

one_sample_t_test <- function(sample,theoretical_mean) {

   ### Returns the t value and degrees of freedom with one sample t test using given sample and theoretical mean. 
   ###	
   ###   Args:
   ###           
   ###	      sample::[List]
   ###	         Given sample 
   ###	      
   ###	      theoretical_mean::[Double]
   ###	         Theoretical mean of sample.
   ###
   ###

   t = (mean(sample,na.rm = TRUE)-theoretical_mean)/(sd(sample, na.rm = TRUE)/(sqrt(length(!is.na(sample)))))
   df = length(sample) - 1 

   return(list(t,df))
}

sample = as.numeric(unlist(gss_data[,"tvhours"])) # "how many hours per day do you watch tv?"
tm = 3.1

result = one_sample_t_test(sample,tm)
result
